tests/infra-validations.sh

```diff
@@ -0,0 +1,103 @@
+
+
+echo "TEST_WORKSPACE: $TEST_WORKSPACE"
+echo "SPK_LOCATION: $SPK_LOCATION"
+echo "AZDO_PROJECT: $AZDO_PROJECT"
+echo "AZDO_ORG: $AZDO_ORG"
+echo "AZDO_ORG_URL: $AZDO_ORG_URL"
+echo "ACR_NAME: $ACR_NAME"
+
+
+terraform_template_dir=discovery-tf-template
+tf_template_version=v0.0.1
+infra_hld_dir=discovery-infra-hld
+services_full_dir="$TEST_WORKSPACE/$mono_repo_dir/$services_dir"
+
+validation_test_yaml="rg_name: <insert value>"
+
+shopt -s expand_aliases
+alias spk=$SPK_LOCATION
+echo "SPK Version: $(spk --version)"
+
+echo "Running from $(pwd)"
+if [ -d "$TEST_WORKSPACE"  ]; then rm -Rf $TEST_WORKSPACE; fi
+
+if [ ! -d "$TEST_WORKSPACE" ]; then
+  echo "Directory '$TEST_WORKSPACE' does not exist"
+  mkdir $TEST_WORKSPACE
+  echo "Created '$TEST_WORKSPACE'"
+fi
+
+cd $TEST_WORKSPACE
+
+# Setup simple TF Tepmlate ------------------
+mkdir $terraform_template_dir
+cd $terraform_template_dir
+git init
+mkdir template
+cd template
+tfTemplate=$'resource "azurerm_resource_group" "example"{\n name= "${var.rg_name}"\n location = "${var.rg_location}"\n}'
+tfVars=$'variable "rg_name" { \n type = "string" \n } \n variable "rg_location" { \n type = "string" \n }'
+touch main.tf variables.tf
+echo "$tfVars" >> variables.tf
+echo "$tfTemplate" >> main.tf
+file_we_expect=("variables.tf" "main.tf")
+validate_directory "$TEST_WORKSPACE/$terraform_template_dir/template" "${file_we_expect[@]}"
+# The TF Template requires a git release for a version to be targeted for spk scaffold
+git add -A
+
+# See if the remote repo exists
+repo_exists $AZDO_ORG_URL $AZDO_PROJECT $terraform_template_dir
+
+# Create the remote repo for the local repo
+created_repo_result=$(az repos create --name "$terraform_template_dir" --org $AZDO_ORG_URL --p $AZDO_PROJECT)
+
+# Extract out remote repo URL from the above result
+remote_repo_url=$(echo $created_repo_result | jq '.remoteUrl' | tr -d '"' )
+echo "The remote_repo_url is $remote_repo_url"
+
+# Remove the user from the URL
+repo_url=$(getHostandPath "$remote_repo_url")
+manifest_repo_url=$repo_url
+
+git commit -m "inital commit"
+git tag "$tf_template_version"
+
+# git remote rm origin
+source=https://infra_account:$ACCESS_TOKEN_SECRET@$repo_url
+git remote add origin "$source"
+echo "git push"
+git push -u origin --all
+git push origin "$tf_template_version"
+
+
+# Scaffold an Infra-HLD repo from TF Template ------------------
+cd ../..
+# Single Cluster scaffold for template
+pwd
+echo "../$terraform_template_dir"
+echo "$tf_template_version"
+spk infra scaffold -n $infra_hld_dir --source "$source" --version "$tf_template_version" --template "template"
+# Validate the definition in the Infra-HLD repo ------------------
+file_we_expect=("definition.yaml")
+validate_directory "$TEST_WORKSPACE/$infra_hld_dir" "${file_we_expect[@]}"
+# Validate the contents of the definition.yaml
+validate_file "$TEST_WORKSPACE/$infra_hld_dir/definition.yaml" $validation_test_yaml
```

Comment:
Since this is a yaml file with particular details in it are we planning to verify further or will the `spa infra generate` tests implicitly verify that?

Reply:
Since the definition.yaml will have the PAT tokens for the source, we are doing a check for tf contents within the inputs block of the yaml. We can expand this later and do more validations with generate.

Reply:
@andrebriggs the `validate_file` function is sufficient in validating the contents of the yaml file. This will be re-used to validate contents in the `spk.tfvars` file in the `spk infra generate` integration test. For now, I think this is sufficient for `spk infra scaffold`.

Reply:
When I looked yesterday the `validate_file` function didn't look like it did that much

Reply:
All the `validate_file` function is doing is scanning a file via `grep` for a particular string. This is how we validate whether a file (e.g. `definition.yaml` or `spk.tfvars`) was scaffolded or generated correctly.

Reply:
@yradsmikham I already I mentioned this to @NathanielRose 
> The check that you have is checking correctly. All that I mean is that we are not doing any sort of YAML file parsing. A grep search would succeed if the file was json internally. 

My assumption is that test for 'spk infra generate' will do that for us. We are on the same page

Reply:
Gotcha, yes, you're correct. If `spk infra generate` is capable of generating a `spk.tfvars` from the `definition.yaml` from scaffold, then this will be validated.